pallets/marketplace/src/lib.rs

```diff
@@ -908,10 +1003,21 @@ pub mod pallet {
                             Ok(())
                         },
                     )?;
-                    <StoredMatchesReverseIndex<T>>::insert(
+                    <AssignedProcessors<T>>::try_mutate(
                         &m.job_id,
-                        planned_execution.source.clone(),
-                    );
+                        |processor| -> Result<(), Error<T>> {
+                            match processor {
+                                None => {
+                                    *processor = Some(Default::default());
```

Comment:
This code is a hard to understand. Should it not insert `planned_execution.source.clone()` into the vec?

This match statement could be replaced with an `if processor.is_none()`